PWGHF/TableProducer/candidateSelectorOmegac0ToOmegaKa.cxx

```diff
@@ -0,0 +1,628 @@
+        resultSelections = false;
+        registry.fill(HIST("hSelDCACharmDau"), 0);
+      } else {
+        registry.fill(HIST("hSelDCACharmDau"), 1);
+      }
+
+      // dcaXY v0 daughters to PV cut
+      if (candidate.dcaXYToPvV0Dau0() < dcaPosToPvMin || candidate.dcaXYToPvV0Dau1() < dcaNegToPvMin) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDcaXYToPvV0Daughters"), 0);
+      } else {
+        registry.fill(HIST("hSelDcaXYToPvV0Daughters"), 1);
+      }
+
+      // dcaXY ka <-- cascade to PV cut
+      if (candidate.dcaXYToPvCascDau() < dcaBachToPvMin) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDcaXYToPvKaFromCasc"), 0);
+      } else {
+        registry.fill(HIST("hSelDcaXYToPvKaFromCasc"), 1);
+      }
+
+      // cut on charm bachelor kaon dcaXY and dcaZ
+      if ((std::abs(candidate.impactParBachFromCharmBaryonXY()) < impactParameterXYKaFromCharmBaryonMin) || (std::abs(candidate.impactParBachFromCharmBaryonXY()) > impactParameterXYKaFromCharmBaryonMax)) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDCAXYPrimKa"), 0);
+      } else {
+        registry.fill(HIST("hSelDCAXYPrimKa"), 1);
+      }
+      if ((std::abs(candidate.impactParBachFromCharmBaryonZ()) < impactParameterZKaFromCharmBaryonMin) || (std::abs(candidate.impactParBachFromCharmBaryonZ()) > impactParameterZKaFromCharmBaryonMax)) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDCAZPrimKa"), 0);
+      } else {
+        registry.fill(HIST("hSelDCAZPrimKa"), 1);
+      }
+
+      // cut on cascade dcaXY and dcaZ
+      if ((std::abs(candidate.impactParCascXY()) < impactParameterXYCascMin) || (std::abs(candidate.impactParCascXY()) > impactParameterXYCascMax)) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDCAXYCasc"), 0);
+      } else {
+        registry.fill(HIST("hSelDCAXYCasc"), 1);
+      }
+      if ((std::abs(candidate.impactParCascZ()) < impactParameterZCascMin) || (std::abs(candidate.impactParCascZ()) > impactParameterZCascMax)) {
+        resultSelections = false;
+        registry.fill(HIST("hSelDCAZCasc"), 0);
+      } else {
+        registry.fill(HIST("hSelDCAZCasc"), 1);
+      }
+
+      // pT selections
+      double ptKaFromCasc = RecoDecay::sqrtSumOfSquares(candidate.pxBachFromCasc(), candidate.pyBachFromCasc());
+      double ptKaFromCharmBaryon = RecoDecay::sqrtSumOfSquares(candidate.pxBachFromCharmBaryon(), candidate.pyBachFromCharmBaryon());
+      if (std::abs(ptKaFromCasc) < ptKaFromCascMin) {
+        resultSelections = false;
+        registry.fill(HIST("hSelPtKaFromCasc"), 0);
+      } else {
+        registry.fill(HIST("hSelPtKaFromCasc"), 1);
+      }
+      if (std::abs(ptKaFromCharmBaryon) < ptKaFromCharmBaryonMin) {
+        resultSelections = false;
+        registry.fill(HIST("hSelPtKaFromCharm"), 0);
+      } else {
+        registry.fill(HIST("hSelPtKaFromCharm"), 1);
+      }
+
+      //  TPC clusters selections
```

Comment:
Hi @fgrosa, thanks for pointing me to this! I will definitely switch to these functions, but in a separate PR, after this one will have been merged (so that I could implement this change also in other files in one go)